ping_test.go

```diff
@@ -0,0 +1,21 @@
+package ping
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRun(t *testing.T) {
+	_, err := Run("asdbasdfasdfdasfkjhasdfkjh.com", nil)
+	assert.Error(t, err)
+
+	stats, err := Run("www.baidu.com", &Opts{
```

Comment:
I'm personally not a fan of tests which rely on an internet connection and/or 3rd party sites functioning as desired.  Kind of a style thing though I guess, so I don't consider it a blocker for this PR.

An easy fix would be something like:

```
s := httptest.NewServer(nil)
defer s.Close()
host, _, _ := net.SplitHostPort(strings.Replace(s.URL, "http://", "", -1))

stats, err := Run(host, &Opts{
    Count: 10,
})
```

(and you could change `Count` to 1)

Reply:
Sadly that wouldn't work because we actually need an ICMP echo server, which requires root.

Reply:
Oh duh.  I tested it out before leaving the comment, but of course it worked because it was just pinging `127.0.0.1`.  I guess that would be a simpler fix - to just ping `localhost`.

All good though.